core/services/ocr2/plugins/vault/service.go

```diff
@@ -108,25 +114,18 @@ func handleRequest[T proto.Message](ctx context.Context, s *Service, id string,
 
 	select {
 	case <-ctx.Done():
-		var zero T
-		return zero, ctx.Err()
+		return nil, ctx.Err()
 	case resp := <-respCh:
-		var zero T
 		if resp.Error != "" {
-			return zero, fmt.Errorf("error processing request %s: %w", id, errors.New(resp.Error))
-		}
-
-		_, ok := resp.Payload.(T)
-		if !ok {
-			return zero, fmt.Errorf("unexpected response type: got %T", resp.Payload)
+			return nil, fmt.Errorf("error processing request %s: %w", id, errors.New(resp.Error))
 		}
 
-		return resp.Payload.(T), nil
+		return resp, nil
 	}
 }
 
-func (s *Service) CreateSecrets(ctx context.Context, request *vault.CreateSecretsRequest) (*vault.CreateSecretsResponse, error) {
-	return handleRequest[*vault.CreateSecretsResponse](ctx, s, request.RequestId, request)
+func (s *Service) CreateSecrets(ctx context.Context, request *vault.CreateSecretsRequest) (*Response, error) {
```

Comment:
@DeividasK @prashantkumar1982 Raising this change here since it breaks the existing interface -- all CRUDL methods facing users will return a Response struct with its Payload field containing the JSON

(Atm this json is a translation of the corresponding proto message, eg. CreateSecretsResponse, but we can tweak it if we want the format to look different to users)